locked-asset/simple-lock-energy/src/unlock_with_penalty.rs

```diff
@@ -197,25 +204,89 @@ pub trait UnlockWithPenaltyModule:
+            }
         }
+
+        // Only once per week
+        self.send_fees_to_collector();
+    }
+
+    /// Merges new fees with existing fees and saves in storage
+    fn merge_fees_from_penalty(&self, token_nonce: Nonce, new_fee_amount: BigUint) {
+        let locked_token_mapper = self.locked_token();
+        let existing_nonce_amount_pair = self.fees_from_penalty_unlocking().get();
+        let mut payments = PaymentsVec::new();
+        payments.push(EsdtTokenPayment::new(
+            locked_token_mapper.get_token_id(),
+            token_nonce,
+            new_fee_amount,
+        ));
+        payments.push(EsdtTokenPayment::new(
+            locked_token_mapper.get_token_id(),
+            existing_nonce_amount_pair.nonce,
+            existing_nonce_amount_pair.amount,
+        ));
+
+        let new_locked_amount_attributes = self.merge_tokens(payments, OptionalValue::None);
+
+        let sft_nonce = self.get_or_create_nonce_for_attributes(
+            &locked_token_mapper,
+            &new_locked_amount_attributes
+                .attributes
+                .original_token_id
+                .clone()
+                .into_name(),
+            &new_locked_amount_attributes.attributes,
+        );
+
+        let new_locked_tokens = locked_token_mapper
+            .nft_add_quantity(sft_nonce, new_locked_amount_attributes.token_amount);
+
+        self.fees_from_penalty_unlocking().set(NonceAmountPair::new(
+            new_locked_tokens.token_nonce,
+            new_locked_tokens.amount,
+        ));
     }
 
-    fn send_fees_to_collector(&self, token_id: TokenIdentifier, amount: BigUint) {
-        let sc_address = self.fees_collector_address().get();
-        self.fees_collector_proxy_builder(sc_address)
-            .deposit_swap_fees()
-            .add_esdt_token_transfer(token_id, 0, amount)
-            .execute_on_dest_context_ignore_result();
+    #[endpoint(sendFeesToCollector)]
+    fn send_fees_to_collector(&self) {
+        // Send fees to FeeCollector SC
+        let current_epoch = self.blockchain().get_block_epoch();
+        let last_week_fee_sent_to_collector = self.last_week_fee_sent_to_collector().get();
+
+        if current_epoch >= last_week_fee_sent_to_collector + EPOCHS_PER_WEEK {
+            let sc_address = self.fees_collector_address().get();
+            let locked_token_id = self.locked_token().get_token_id();
+            let nonce_amount_pair = self.fees_from_penalty_unlocking().get();
+
+            self.fees_from_penalty_unlocking().clear();
+            self.fees_collector_proxy_builder(sc_address)
+                .deposit_swap_fees()
+                .add_esdt_token_transfer(
+                    locked_token_id,
+                    nonce_amount_pair.nonce,
+                    nonce_amount_pair.amount,
+                )
+                .execute_on_dest_context_ignore_result();
+
+            self.last_week_fee_sent_to_collector().set(current_epoch);
+        }
```

Comment:
It's usually better to have early exit than everything in an `if`. Can rewrite as:
```
let next_send_epoch = last_week_fee_sent_to_collector + EPOCHS_PER_WEEK;
if current_epoch < next_send_epoch {
     return;
}

// code here